pkg/services/services.go

```diff
@@ -210,6 +210,18 @@ func getKubeletServiceConfiguration(argsFromIginition map[string]string, debug b
 		// special case substitution handled in WICD itself
 		kubeletServiceCmd = fmt.Sprintf("%s --node-ip=%s", kubeletServiceCmd, NodeIPVar)
 	}
+
+	// explicitly set node ip and resolves to the first IPv4 address of the default gateway
+	kubeletServiceCmd = fmt.Sprintf("%s --node-ip=%s", kubeletServiceCmd, NodeIPVar)
+	if platform == config.AWSPlatformType {
+		kubeletServiceCmd = fmt.Sprintf("%s --image-credential-provider-bin-dir=%s --image-credential-provider-config=%s",
+			kubeletServiceCmd, windows.K8sDir, windows.CredentialProviderConfig)
+	}
+	preScripts = append(preScripts, servicescm.PowershellPreScript{
+		VariableName: NodeIPVar,
+		Path: "(Get-NetRoute -DestinationPrefix '0.0.0.0/0' | " +
+			"Get-NetIpAddress -AddressFamily IPv4 -ifIndex {$_.ifIndex}[0]).IPAddress",
+	})
```

Comment:
This looks like it's coming from https://github.com/sebsoto/windows-machine-config-operator/commit/629fd94635a22cf044a049a0dc760b7819c245bc -- if we include this, do we also want to remove the WICD logic around replacing NodeIPVar? Tracing the code, I don't think it would affect functionality but would remove unneeded code